mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -166,6 +169,69 @@ def __init__(self, name=None, bus=None, use_settings=True):
         # Skill Public API
         self.public_api = {}
 
+    @property
+    def is_fully_initialized(self):
+        return self._init_event.is_set()
+
+    def handle_first_run(self):
+        """The very first time a skill is run, speak the intro."""
+        intro = self.get_intro_message()
+        if intro:
+            # supports .dialog files for easy localization
+            # when .dialog does not exist, the text is spoken
+            # it is backwards compatible
+            self.speak_dialog(intro)
+
+    def _check_for_first_run(self):
+        """Determine if its the very first time a skill is run."""
+        first_run = self.settings.get("__mycroft_skill_firstrun", True)
+        if first_run:
+            LOG.info("First run of " + self.skill_id)
+            self.handle_first_run()
+            self.settings["__mycroft_skill_firstrun"] = False
+            save_settings(self.settings_write_path, self.settings)
+
+    def _startup(self, bus, skill_id=""):
+        """Startup the skill.
+
+        This connects the skill to the messagebus, loads vocabularies and
+        data files and in the end calls the skill creator's "intialize" code.
+
+        Arguments:
+            bus: Mycroft Messagebus connection object.
+            skill_id (str): need to be unique, by default is set from skill path
+                but skill loader can override this
+        """
+        if self.is_fully_initialized:
+            return  # already initialized
+        # NOTE: this method is called by SkillLoader
+        # it is private to make it clear to skill devs they should not touch it
+        try:
+            # set the skill_id
+            self.skill_id = skill_id or basename(self.root_dir)
+            self.intent_service.set_id(self.skill_id)
+            self.event_scheduler.set_id(self.skill_id)
+            self._init_settings()
+
+            # initialize anything that depends on the messagebus
+            self.bind(bus)
+            self.load_data_files()
+            self._register_decorated()
+            self.register_resting_screen()
+
+            # run skill developer initialization code
+            self.initialize()
+            self._check_for_first_run()
+            self._init_event.set()
+        except Exception as e:
+            LOG.exception('Skill initialization failed')
+            # If an exception occurs, attempt to clean up the skill
+            try:
+                self.default_shutdown()
+            except Exception as e2:
+                pass
```

Comment:
it should not, because it happens 99% of the time, its just useless spam, we expect this to fail often in skills that implement their own shutdown due to accessing variables that havent been set etc,

this was also just moved from other file and not changed